python/tank/util/metrics.py

```diff
@@ -533,25 +533,59 @@ def is_supported_event(self):
         return repr(self) in EventMetric.SUPPORTED_EVENTS
 
     @classmethod
-    def log(cls, group, name, properties=None, log_once=False):
+    def log(cls, group, name, properties=None, log_once=False, context=None):
         """
         Queue a metric event with the given name for the given group on
         the :class:`MetricsQueueSingleton` dispatch queue.
 
-        This method simply adds the metric event to the dispatch queue meaning that 
+        This method simply adds the metric event to the dispatch queue meaning that
         the metric has to be treated by a dispatcher to be posted.
 
         :param str group: A group or category this metric event falls into.
                           Any values can be used to implement your own taxonomy,
                           the "Toolkit" group name is reserved for internal use.
         :param str name: A short descriptive event name or performed action,
                          e.g. 'Launched Command', 'Opened Workfile', etc..
-        :param dict properties: An optional dictionary of extra properties to be 
+        :param dict properties: An optional dictionary of extra properties to be
                                 attached to the metric event.
         :param bool log_once: ``True`` if this metric should be ignored if it has
                               already been logged. Defaults to ``False``.
-
+        :param <TankBundle> context: A `TankBundle` based class e.g.:app, engine or framework
         """
+
+        base_properties = None
+
+        if not context:
+            # No context specified, try guessing one
+            try:
+                # import here to prevent circular dependency
+                from ..platform.util import current_bundle
```

Comment:
Use `from sgtk.platform.util import current_bundle` instead of relative imports.